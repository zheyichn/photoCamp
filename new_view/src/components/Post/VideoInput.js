import React from "react";

export default function VideoInput(props) {
  const { height = 300, source } = props;
  return (
    <div className="VideoInput">
      {source && (
        <video
          className="VideoInput_video"
          width="100%"
          height={height}
          controls
          src={source}
        />
      )}
      <div className="VideoInput_footer" data-testid="video-footer">
        {source || "Nothing selectd, please refresh page and select again"}
      </div>
    </div>
  );
}
