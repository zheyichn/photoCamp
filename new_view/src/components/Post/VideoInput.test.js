/**
 * @jest-environment jsdom
 */
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup, screen } from "@testing-library/react";
// import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import VideoInput from "./VideoInput";

// afterEach function runs after each test suite is executed
afterEach(() => {
  cleanup();
});

describe("VideoInput Component", () => {
  const testVideoSource =
    "https://pixabay.com/videos/grass-dew-field-fog-morning-66810/";

  //Test1
  test("show video link when video selected", () => {
    render(<VideoInput source={testVideoSource} />);
    const footerDiv = screen.getByTestId("video-footer");
    expect(footerDiv).toBeInTheDocument();
    expect(footerDiv).toHaveTextContent(testVideoSource);
  });

  // Test2: match snapshots
  it("VideoInput renders correctly", () => {
    const tree = renderer
      .create(<VideoInput source={testVideoSource} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
