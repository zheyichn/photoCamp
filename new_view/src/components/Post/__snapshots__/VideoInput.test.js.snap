// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInput Component VideoInput renders correctly 1`] = `
<div
  className="VideoInput"
>
  <video
    className="VideoInput_video"
    controls={true}
    height={300}
    src="https://pixabay.com/videos/grass-dew-field-fog-morning-66810/"
    width="100%"
  />
  <div
    className="VideoInput_footer"
    data-testid="video-footer"
  >
    https://pixabay.com/videos/grass-dew-field-fog-morning-66810/
  </div>
</div>
`;
